gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -163,6 +182,13 @@ String generateSpecifiedPartitions(Histogram histogram, int maxPartitions, long
       }
     }
 
+    if (partitionPoints.isEmpty()) {
+      throw new RuntimeException("Unexpected empty partition list");
+    }
+
+    // exchange the first partition point with the global low watermark
+    partitionPoints.set(0, Long.toString(lowWatermark));
```

Comment:
Instead of doing this, we can set the first `HistogramGroup` key to be lowWatermak, which also helps with fine probing.